### PR TITLE
implement Serialize/Deserialize for SystemTime with RFC3339 format

### LINKED ISSUE #7151: pageserver: implement Serialize for SystemTime with RFC3339 format
Since https://github.com/neondatabase/neon/pull/7131 we have two places that use a helper (`ser_rfc3339_millis`) to get serde to string-ize SystemTimes into the format we want.

It would be nice to just have a Serialize implementation so that we don't have to decorate structs.

### PR DESCRIPTION
**Pull request by kevinmingtarja**
## Problem
We have two places that use a helper (`ser_rfc3339_millis`) to get serde to stringify SystemTimes into the desired format. 

## Summary of changes
Created a new module `utils::serde_system_time` and inside it a wrapper type `SystemTime` for `std::time::SystemTime` that serializes/deserializes to the RFC3339 format.

This new type is then used in the two places that were previously using the helper for serialization, thereby eliminating the need to decorate structs.

Closes #7151.

### CODE REVIEW

**Comment by jcsp** [comment]
Thanks Kevin - @VladLazar could you review please?

File: libs/utils/src/serde_system_time.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+//! A `serde::{Deserialize,Serialize}` type for SystemTime.
```
Comment (VladLazar) [inline libs/utils/src/serde_system_time.rs:1]: Expanding this comment to mention the format serialization format or changing the struct name to reflect that would be nice.

**Review by VladLazar — APPROVED** [review]
Thanks for the PR. Looks good to me (just one minor nit).

File: libs/utils/src/serde_system_time.rs (hunk 1 above)
Comment (kevinmingtarja) [reply libs/utils/src/serde_system_time.rs]: Got it, I've improved the comment.

**Review by VladLazar — COMMENTED** [review]
@kevinmingtarja looks like your PR raced with a change to the utilization endpoint. Could you please update your branch? I'll run CI again after that and merge it.

**Comment by kevinmingtarja** [comment]
> @kevinmingtarja looks like your PR raced with a change to the utilization endpoint. Could you please update your branch? I'll run CI again after that and merge it.

Got it, I just updated my branch and resolved the merge conflict.

**Comment by VladLazar** [comment]
@kevinmingtarja thanks for the PR!

**Pull request merged**